src/twisted/names/test/test_names.py

```diff
@@ -594,6 +595,28 @@ def test_recordMissing(self):
         self.assertEqual(additional, [])
 
 
+    def test_unknownTypeNXDOMAIN(self):
+        """
+        Requesting a record of unknown type where no records exist for the name
+        in question results in L{DomainError}.
+        """
+        testDomain = test_domain_com
+        testDomainName = b'nonexistent.prefix-' + testDomain.soa[0]
+        f = self.failureResultOf(
+            testDomain.query(Query(name=testDomainName, type=76)))
```

Comment:
The maximum supported type in Twisted is 255 (ALL_RECORDS), so this would produce an illegal value (256). Oh, this is a 16-bit value, not an 8-bit value. Hmm. I guess that would work, then.